### PR TITLE
feat: add title, summary and description to the pipeline model

### PR DESCRIPTION
**Pull request by morbalint**
BIVS-1969

### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-io/bitrise/blob/master/.github/CONTRIBUTING.md)
- [x] `README.md` is updated with the changes (if needed)

### Version

Requires a *MINOR* [version update](https://semver.org/)

### Context

More human readable description of pipelines and workflow to enable adding more visibility to them, by adding a title, summary and description field to pipeline elements. Registration of these fields is required so that the WFE workflows tab save button doesn't remove them from an existing bitrise.yml.

### Changes

Adds `title`, `summary` and `description` to the [PipelineModel](https://github.com/bitrise-io/bitrise/blob/d2d4706931314efcc7cda16bf9b7a9ca04a0e90e/models/models.go#L21) and [StageModel](https://github.com/bitrise-io/bitrise/blob/d2d4706931314efcc7cda16bf9b7a9ca04a0e90e/models/models.go#L29)

### Investigation details

When editing the bitrise.yml in the visual WFE, the save button on the workflows tab removes all unknown fields 😞

### CODE REVIEW

File: models/models.go [hunk 1]
```diff
@@ -19,7 +19,10 @@ type StepListItemModel map[string]stepmanModels.StepModel
 
 // PipelineModel ...
 type PipelineModel struct {
-	Stages []StageListItemModel `json:"stages,omitempty" yaml:"stages,omitempty"`
+	Stages      []StageListItemModel `json:"stages,omitempty" yaml:"stages,omitempty"`
+	Title       string               `json:"title,omitempty" yaml:"title,omitempty"`
```
Comment (godrei) [inline models/models.go:23]: Pls move `Title`, `Summary` and `Description` to the top of the struct fields, to follow the convention.

**Review by godrei — CHANGES_REQUESTED** [review]

**Comment by godrei** [comment]
Pls fill in the PR template to let others understand this change in the future.

File: models/models.go [hunk 2]
```diff
@@ -19,7 +19,10 @@ type StepListItemModel map[string]stepmanModels.StepModel
...
+	Summary     string               `json:"summary,omitempty" yaml:"summary,omitempty"`
+	Description string               `json:"description,omitempty" yaml:"description,omitempty"`
```
Comment (morbalint) [inline models/models.go:25]:
```suggestion
	Title       string               `json:"title,omitempty" yaml:"title,omitempty"`
	Summary     string               `json:"summary,omitempty" yaml:"summary,omitempty"`
	Description string               `json:"description,omitempty" yaml:"description,omitempty"`
	Stages      []StageListItemModel `json:"stages,omitempty" yaml:"stages,omitempty"`
```

File: models/models.go (hunk 1 above)
Comment (morbalint) [reply models/models.go:23]: done

**Review by godrei — APPROVED** [review]

**Pull request merged**